stablehlo/dialect/StablehloOps.cpp

```diff
@@ -4660,49 +4661,65 @@ LogicalResult ReplicaIdOp::inferReturnTypes(
 // If Op
 //===----------------------------------------------------------------------===//
 
-static LogicalResult verifyConditionalBranch(Operation* op, Region& region,
-                                             llvm::Twine branchName) {
-  if (region.getNumArguments() != 0)
-    return op->emitOpError()
-           << branchName << " must have 0 arguments, but found "
-           << region.getNumArguments();
-
-  TypeRange branchReturnTypes =
-      region.front().getTerminator()->getOperandTypes();
-  if (branchReturnTypes != op->getResultTypes())
-    return op->emitOpError()
-           << branchName << " returned types (" << branchReturnTypes
-           << ") do not match op result types (" << op->getResultTypes() << ")";
+// TODO: remove after https://github.com/openxla/stablehlo/pull/145 landed.
```

Comment:
Given that this PR will likely land before the mentioned PR, let's add this function to Base.h right here?

Reply:
Yes, I copy `isCompatibleForHloTypeInference` from`Base.h/cc`in https://github.com/openxla/stablehlo/pull/145 to the same location here in this PR.